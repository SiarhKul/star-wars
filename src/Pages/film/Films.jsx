import "react-toastify/dist/ReactToastify.css";

import React from "react";
import { Cards } from "../../components/cards/Cards";
import { ToastContainer } from "react-toastify";
import { bodyCardFilm } from "../../components/bodyCards/bodyCardFIlm";
import { useSelector } from "react-redux";
import { Loader } from "../../components/loader/Loader";
import { history } from "../../redux/reducers";
import { selectFilmsStore } from "../../redux/selectors/selectors";

export const Films = () => {
	const contenCards = useSelector(selectFilmsStore);
	const { pathname } = history.location;

	return (
		<main>
			<Loader />
			<ToastContainer />
			<Cards
				name="title"
				path={pathname}
				contenCards={contenCards}
				BodyComponent={bodyCardFilm}
			/>
		</main>
	);
};
