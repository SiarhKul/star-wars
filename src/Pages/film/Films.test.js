import React from "react";
import { Films } from "./Films.jsx";
import * as redux from "react-redux";

jest.mock("history", () => {
	return {
		createBrowserHistory: jest.fn(() => ({
			location: {
				pathname: "/films",
			},
		})),
	};
});

describe("Test <Films/>", () => {
	const films = () => shallow(<Films />);
	let component;
	let spyOnUseSelector;

	beforeEach(() => {
		spyOnUseSelector = jest.spyOn(redux, "useSelector").mockReturnValue([{}]);
		component = films();
	});

	it("render snapshot <Films/>", () => {
		expect(component).toMatchSnapshot();
	});

	it("should render <Films/>", () => {
		expect(component.find("Loader")).toHaveLength(1);
	});

	it("should render alert message in <Films/>", () => {
		expect(component.find("ToastContainer")).toHaveLength(1);
	});

	it("should render Cards component in <Films/>", () => {
		expect(component.find("Cards")).toHaveLength(1);
	});
});
