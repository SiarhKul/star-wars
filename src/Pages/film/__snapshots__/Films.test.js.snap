// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Films/> render snapshot <Films/> 1`] = `
<main>
  <Loader />
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggableDirection="x"
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    transition={[Function]}
  />
  <Cards
    BodyComponent={[Function]}
    contenCards={
      Array [
        Object {},
      ]
    }
    name="title"
    path="/films"
  />
</main>
`;
